src/encodings/decoding.rs

```diff
@@ -428,25 +407,55 @@ impl Decoder<Int64Type> for DeltaBitPackDecoder<Int64Type> {
         }
       }
 
-      // we can't move this outside the loop since it needs to
-      // "carve out" the permission of `self.bit_reader` which makes
-      // `self.init_block()` call impossible (the latter requires the
-      // whole permission of `self`)
-      // TODO: evaluate the performance impact of this.
-      let bit_reader = self.bit_reader.as_mut().unwrap();
-
       // TODO: use SIMD to optimize this?
-      let delta = bit_reader.get_value::<u64>(self.delta_bit_width as usize)
+      let delta = self.bit_reader.get_value::<u64>(self.delta_bit_width as usize)
         .ok_or(eof_err!("Not enough data to decode 'delta'"))?;
-      self.current_value += self.min_delta;
-      self.current_value += delta as i64;
-      buffer[i] = self.current_value;
+      self.current_value = self.current_value.wrapping_add(self.min_delta);
```

Comment:
This is done, because it is okay for values to overflow: [parquet-mr comment](https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/column/values/delta/DeltaBinaryPackingValuesWriterForLong.java#L89). We should still process current value and/or delta in either decoding or encoding.

For example, this can happen when you have a sequence of i64::MAX and i64::MIN values - the test `encodings::decoding::tests::test_delta_bit_packed_int64_min_max` fails if one changes the code to remove wrapping; however, I think it is perfectly valid to process such sequence of values for decoding.

I am happy to make the change, but then I will have to update tests too.

Reply:
Ah - make sense. I guess it is OK for deltas to contain "overflowed" values as long as `wrapping_add` can restore the original value. Can we add some comments for this code?

Reply:
Will add, thanks!